lisa/wlgen/rta.py

```diff
@@ -677,11 +741,34 @@ def __init__(self, start_pct=100, end_pct=0, time_s=1, period_ms=100,
         if end_pct >= start_pct:
             raise ValueError('end_pct must be lower than start_pct')
 
+        # Clamp values for previous phases.
+        # These values are remembered so the clamp values are not set on the start
+        #   of every phase
+        # The initial values should probably be set on the first phase,
+        #   this protects against retaining a clamp from a previous loop.
+        clamp_min = None
+        clamp_max = None
+
         phases = []
-        for load in [start_pct, end_pct]:
+        for idx, load in enumerate([start_pct, end_pct]):
             if load == 0:
                 continue
+                
+            # If clamp values are unchanged, don't bother setting them again
+            new_clamp_min = clamp_mins(idx, load)
```

Comment:
Since the exact same pattern came up twice, you could factor it out in a function like this one:
```
import itertools

def replace_consecutive_values(l, placeholder=None):
    """
    Replace consecutive runs of the same value with ``placeholder``.
    """
    if len(l) < 2:
        return list(l)
    prev_val = l[0]
    deduplicated = []
    for val in l[1:]:
        if val == prev_val:
            dedup_val = None
        else:
            dedup_val = val
            prev_val = val

        deduplicated.append(dedup_val)
    return deduplicated

def compute_clamps(steps, get_clamps):
    if not get_clamps:
        get_clamps = lambda idx, load: (None, None)

    # Use starmap instead of map to unpack the tuple from enumerate
    clamps = starmap(get_clamps, enumerate(steps))
    # Unzip the clamp tuples into two separate iterables
    both_clamps = zip(*clamps)
    # Remove consecutive runs of the same clamp value
    both_clamps = map(replace_consecutive_values, both_clamps)
    # Zip back into a single list of tuples
    return list(zip(*both_clamps))
```
And then use it as:
```
# get_clamps is either None or a function returning (clamp_min, clamp_max)
clamps = compute_clamps(steps, get_clamps)
for load, (min_clamp, max_clamp) in zip(steps, clamps):
    ...
```

Note: `replace_consecutive_values` may benefit from being moved to `lisa/utils.py`, as it is generally useful (there are some other itertools-based helpers there).